scripts/serie_update.py

```diff
@@ -0,0 +1,674 @@
+#!/usr/bin/python
+"""
+SPDX-License-Identifier: Apache-2.0
+Copyright (c) 2019 STMicroelectronics.
+This script define Stm32SerieUpdate class
+to be used by update_stm32_package.py
+"""
+
+import os
+import stat
+import shutil
+import subprocess
+import re
+from pathlib import Path
+import logging
+
+STM32_CUBE_REPO_BASE = "https://github.com/STMicroelectronics/STM32Cube"
+
+
+#################################
+# Remove read only protection
+#
+def remove_readonly(func, path, _):
+    os.chmod(path, stat.S_IWRITE)
+    func(path)
+
+
+#################################
+# Execute a command from shell
+#
+def os_cmd(cmd):
+    logging.info(cmd)
+    return subprocess.call(cmd, shell=True)
+
+
+#################################
+# class Stm32SerieUpdate
+#
+class Stm32SerieUpdate:
+    def __init__(self, stm32_serie, stm32cube_repo_path, force, noclean):
+        if not stm32_serie.startswith("stm32"):
+            raise Exception(
+                "Error: Unknown stm32 serie: "
+                + stm32_serie
+                + ". Must start with 'stm32'"
+            )
+
+        # Set serie variables
+        self.stm32_serie = stm32_serie
+        self.stm32_seriexx = stm32_serie + "xx"  # ex:stm32f3xx
+        self.stm32_serie_upper = stm32_serie.upper()  # ex:STM32F3
+        self.stm32_seriexx_upper = self.stm32_serie_upper + "xx"  # ex:STM32F3xx
+        self.serie = self.stm32_serie_upper[5:]
+        self.force = force
+        self.noclean = noclean
+
+        # #####  3 root directories to work with ########
+        # 1: STM32Cube repo Default $HOME/STM32Cube_repo
+        # 2 : zephyr stm32 path  : ex: .../zephyr_project/module/hal/stm32
+        # 3: Temporary directory to construct the update
+        # (within STM32Cube repo dir)
+        self.stm32cube_repo_path = stm32cube_repo_path
+        if not self.stm32cube_repo_path.exists():
+            self.stm32cube_repo_path.mkdir()
+
+        self.zephyr_hal_stm32_path = (
+            Path(os.getenv("ZEPHYR_BASE")).absolute()
+            / ".."
+            / "modules"
+            / "hal"
+            / "stm32"
+        )
+        if not self.zephyr_hal_stm32_path.exists():
+            raise Exception("Error: cannot find zephyr project")
```

Comment:
Generic `Exception` should be avoided. Maybe `FileNotFoundError` would be suitable here.

Reply:
Done